server/aesdsocket.c

```diff
@@ -0,0 +1,218 @@
+#include<sys/stat.h>
+
+
+#define port 9000
+
+//using global for handlling signals
+int socFD, client_socFD, fd;
+
+static void sigintHandler(int sig)
+{
+   syslog(LOG_NOTICE, "Caught signal, exiting");
+   close(fd);
+   remove("/var/tmp/aesdsocketdata");   //Deleting the file  from fs
+   close(socFD);                        //and closing the open sockets
+   close(client_socFD);
+   
+}
+
+int main()
+{
+
+  openlog(NULL, 0, LOG_USER);
+  char s[INET6_ADDRSTRLEN];
+  int rc;
+
+  //Open the server socket
+  socFD= socket(AF_INET,SOCK_STREAM,0);
+  if(socFD == -1){
+   perror("create socket:");
+   return -1;
+  }
+  
+  //using sockaddr_in struct to hold address both the server and client
+  struct sockaddr_in server, client;
+  socklen_t Client_len;
+  
+  
+  memset((void *)&server,0, sizeof(server));
+  server.sin_family=AF_INET;
+  server.sin_addr.s_addr=INADDR_ANY;
+  server.sin_port=htons(port);
+ 
+  // Adding Socket options to allow reuse of Socket
+  // reference: Beej's guide
+  int yes=1;
+  if (setsockopt(socFD,SOL_SOCKET,SO_REUSEADDR,&yes,sizeof yes) == -1) {
+    perror("setsockopt");
+    return -1;
+    }
+    
+
+
+  //Binding  
+  rc=bind(socFD, (struct sockaddr *)&server, sizeof(server));
+  if(rc == -1)
+  {
+    close(socFD);
+    printf("Error binding: \n");
+    perror("bind error:");
+    return -1;
+  }
+
+
+    // forking to run the program as daemon
+    pid_t pd;
+    pd=fork();
+    
+    if(pd<0)
+      return -1;
+    else if(pd>0){
+      printf("%d", pd);
+      return 0;
+      }
+      
+    if( setsid() <0)
+     return -1;
+     
+     
+   signal(SIGCHLD, SIG_IGN);
+   signal(SIGHUP, SIG_IGN);
+ 
+  
+   umask(0);
+   chdir("/");
+    
+    close(STDIN_FILENO);
+    close(STDOUT_FILENO);
+    close(STDERR_FILENO); 
+
+ 
+    if(signal(SIGINT | SIGTERM, sigintHandler) == SIG_ERR);
+
+    char Data_Byte;
+    int count=0;
+    //int fd;
+ 
+     //opening up file in both read and write mode
+     fd= open("/var/tmp/aesdsocketdata", O_TRUNC|O_CREAT|O_RDWR, 0777);
+     if(fd<0)
+      printf("error opening the file:\n");
+      
+
+      
+ while(1){
+     
+     rc=listen(socFD, 5);   
+     if(rc == -1)
+     {
+          perror("listen:");
+          return (-1);
+     }
+     
+     
+     Client_len = sizeof(client);
+     client_socFD = accept(socFD, (struct sockaddr *)&client, &(Client_len));
+     //error conditionss 
+     if(client_socFD == -1)
+     {
+       perror("accept:");
+       return(-1);
+     }
+     
+     //printing IP address
+     inet_ntop(client.sin_family, (struct sockaddr *)&client,s,sizeof s );
+     //printf("server: got connection from %s\n", s);
+     syslog(LOG_NOTICE, "Accepted connection from %s", s);
+
+     char * str_to_append = (char *) malloc(sizeof(char)) ;
+     int realloc_flag;
+     int send_file=0;
+    
+    realloc_flag=0;
+   
+   while(!send_file) 
+   {  
+     if(realloc_flag)
+     {
+       str_to_append=realloc(str_to_append, count+1);
```

Comment:
@jasan2303 what assumption is made here about the size of RAM heap relative to the total size of all packets sent?